lua/terrortown/menus/gamemode/roles/base_roles.lua

```diff
@@ -167,3 +167,8 @@ function CLGAMEMODESUBMENU:Populate(parent)
 
     hook.Run("TTT2OnRoleAddToSettingsMenu", self.roleData, parent)
 end
+
+-- Override this here, because we don't want auto-discovery to show this menu
+function CLGAMEMODESUBMENU:ShouldShow()
+    return false
+end
```

Comment:
I'm not 100% sure what you're doing here and why. What is "auto-discovery"? What is the reason why this had to be changed from how it was before?

Reply:
Since the roles menu now contains normal, non-generated submenus, the *base* for the submenu would get picked up and shown *as a non-generated submenu*, which is undesirable. We set this here so that it doesn't do that. The virtual submenu logic is otherwise kept exactly.